peripherals/cordic/cordic_g4.yaml

```
@@ -0,0 +1,50 @@
+# CORDIC coprocessor provides hardware acceleration
+# for computing math funcitons
+
+CORDIC:
+  CSR:
+    RRDY:
+      _read:
+        NotReady: [0, "Results from computation are not read"]
+        Ready: [1, "Results are ready, this flag will be automatically cleared once value is read"]
+      _write:
+        Clear: [0, "Clear the computation results"]
+    ARGSIZE:
+      Size32: [0, "Use 32 bit input values"]
+      Size16: [1, "Use 16 bit input values"]
+    RESSIZE:
+      Size32: [0, "Use 32 bit output values"]
+      Size16: [1, "Use 16 bit output values"]
+    NARGS:
+      Num1: [0, "Only single argument write is needed for next calculation"]
+      Num2: [1, "Two argument writes need to be performed for next calculation"]
+    NRES:
+      Num1: [0, "Only single result value will be returned. After a single read RRDY will be automatically cleared"]
+      Num2: [1, "Two return reads need to be performed. After two reads RRDY will be automatically cleared"]
+    DMAWEN:
+      Disabled: [0, "No DMA channel writes are generated"]
+      Enabled: [1, "Write requests are generated on the DMA channel when no operation is pending"]
+    DMAREN:
+      Disabled: [0, "No DMA channel reads are generated"]
+      Enabled: [1, "Read requests are generated on the DMA channel when RRDY flag is set"]
+    IEN:
+      Disabled: [0, "Disable interrupt request generation"]
+      Enabled: [1, "Enable intterrupt request generation"]
+    PRECISION: [1, 15]
+    SCALE: [0, 3]
```

```suggestion
    SCALE: [0, 7]
```

As far as I could tell SCALE can use all 3 bits, so 0 to 7.

---

Good catch - just double checked. My brain forgot to do `2^n-1`